test/unit/plugins/rest_spec.js

```diff
@@ -646,4 +647,97 @@ describe("The Rest plugin", function () {
 			});
 		});
 	});
+
+	describe("updating a created instance", function () {
+
+		var server    = new Hapi.Server();
+		var instances = new InstanceAdapter();
+		var result;
+		var instanceID;
+		var updatedInstance;
+
+		before(function () {
+
+			server.connection();
+			return server.registerAsync({
+				register : Rest,
+				options  : {
+					instances : instances
+				}
+			});
+		});
+
+		after(function () {
+			return server.stopAsync();
+		});
+
+		describe("setting the status of a created instance to failed", function () {
+
+			before(function () {
+				return instances.createInstance()
+				.then(function (instance) {
+					instanceID = instance.id;
+					updatedInstance = new Instance({
+						id    : instance.id,
+						ami   : instance.ami,
+						type  : instance.type,
+						state : "failed",
+						uri   : instance.uri
+					});
+
+					var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+					.payload({
+						ami   : instance.ami,
+						type  : instance.type,
+						uri   : instance.uri,
+						state : "failed"
+					});
+					return request.inject(server);
+				})
+				.then(function (response) {
+					console.log(response);
```

Comment:
This dirties up the logs in the CI system. We should remove this before merging this code.